hatchling-web-app/src/components/TextEditor.jsx

```diff
@@ -57,6 +57,34 @@ export default function Texteditor() {
     }
   }
 
+  function onDownloadButtonClick() {
+    const fileName = prompt("Enter File Name");
+
+    if (fileName === null) {
```

Comment:
Add a console.log to see if this case will ever happen. If so, display the same alert below and return from the function.